<?xml version="1.0" ?>

<launch>
  <arg name="namespace" default="uav"/>
  <arg name="rate" default="100"/>
  <arg name="x" default="0.0"/>
  <arg name="y" default="0.0"/>
  <arg name="z" default="0.0"/>
  <arg name="run_trajectory_node" default="true" />

  <group ns="$(arg namespace)">
    <!-- Attitude control -->
    <node name="rotors_variation_attitude_control" pkg="mmuav_control" type="rotors_variation_attitude_control.py">
      <param name="rate" value="$(arg rate)"/>
    </node>

    <!-- Position control -->
    <node name="position_control" pkg="mmuav_control" type="position_ctl.py" output="screen">
      <param name="rate" value="$(arg rate)"/>
      <param name="filename" value="UavPositionControl.yaml"/>
      <param name="x" value="$(arg x)"/>
      <param name="y" value="$(arg y)"/>
      <param name="z" value="$(arg z)"/>
    </node>

    <!-- Merge position and attitude control node -->
    <node name="rotors_variation_controller_outputs_to_motor_velocities" pkg="mmuav_control" type="rotors_variation_controller_outputs_to_motor_velocities.py">
      <param name="rate" value="$(arg rate)"/>
    </node>

    <!-- Run the trajectory reference node -->
    <group if="$(arg run_trajectory_node)">
      <node name="trajectory_to_trajectory_point" pkg="mmuav_control" type="trajectory_to_trajectory_point.py">
        <param name="rate" value="$(arg rate)"/>
      </node>
    </group>

  </group>

</launch>
